src/utils/sync.ts

```diff
@@ -0,0 +1,44 @@
+/* Mutex provides mutex feature by building a promise chain.
+  const m = new Mutex();
+  await m.withLock(async () => {
+    // critical section
+  })
+
+  https://github.com/golang/vscode-go/blob/fa820d45d7b9217e949c21ffd13268e20c7e0ee4/src/utils/mutex.ts#L8
+*/
+export class Mutex {
+	private mutex = Promise.resolve()
```

Comment:
Can we format the file, seems like the indentation is sometimes 8 and sometimes 2